src/method_nlar.jl

```diff
@@ -7,151 +7,84 @@ export default_eigval_sorter
-            end
-        end
-    end
-
-    #Compute residuals for each Ritz value
-    temp_res = 0;
-    for i=1:size(dd,1)
-        temp_res = errmeasure(dd[i],Vk*vv[:,i]);
-        if(temp_res > threshold)
-            eig_res[i] = threshold;
-        else
-            eig_res[i] = temp_res;
-        end
-    end
-
-    #Sort according to methods
-    ii = sortperm(eig_res.*abs.(dd2-σ));
-
-    mm_min = min(mm,length(ii));
-
-    nu = dd[ii[1:mm_min]];
-    y = vv[:,ii[1:mm_min]];
-
-    return nu,y;
-end
+"""
+    function nlar([eltype],nep::ProjectableNEP,[orthmethod=ModifiedGramSchmidt],[nev=10],[errmeasure=default_errmeasure],[tol=eps(real(T))*100],[maxit=100],[λ0=0],[v0=randn(T,size(nep,1))],[displaylevel=0],[linsolvercreator=default_linsolvercreator],[R=0.01],[eigval_sorter=residual_eigval_sorter],[qrfact_orth=false],[max_subspace=100],[num_restart_ritz_vecs=8],[inner_solver_method=NEPSolver.DefaultInnerSolver])
+ 
+The function implements the Nonlinear Arnoldi method, which finds `nev` eigenpairs(or throws a `NoConvergenceException`) by projecting the problem to a subspace that is expanded in the course  of the algorithm. 
+The basis is orthogonalized either by using the QR method if `qrfact_orth` is `true` or else by an orthogonalization method `orthmethod`). 
+This entails solving a smaller projected problem using a method specified by `inner_solver_method`. 
+(`λ0`,`v0`) is the initial guess for the eigenpair. `linsolvercreator` specifies how the linear system is created and solved. 
+`R` is a parameter used by the function specified by `eigval_sorter` to reject those ritz values that are within a distance `R` from any of the converged eigenvalues, so that repeated convergence to the same eigenpair can be avoided. 
+`max_subspace` is the maximum allowable size of the basis befor the algorithm restarts using a basis made of `num_restart_ritz_vecs` ritz vectors and the eigenvectors that the algorithm has converged to.
+
+# Example
+```julia-repl
+julia> nep=nep_gallery("dep0_tridiag");
+julia> λ,v=nlar(nep,tol=1e-5,nev=1,maxit=50);
+julia> norm(compute_Mlincomb(nep,λ[1],v))
+7.722757003764154e-7
+```
+
+# References
+* H. Voss, An Arnoldi method for nonlinear eigenvalue problems. BIT. Numer. Math. 44: 387-401, 2004.
 
+"""
 nlar(nep::NEP;params...) = nlar(ComplexF64,nep::NEP;params...)
 function nlar(::Type{T},
             nep::ProjectableNEP;
             orthmethod::Type{T_orth} = ModifiedGramSchmidt,
             nev::Int=10,                                     #Number of eigenvalues required
-            submax_rest::Int=30,                             #Maximum dimension of subspace before restarting
             errmeasure::Function = default_errmeasure(nep),
             tol = eps(real(T))*100,
             maxit::Int = 100,
-            λ0 = zero(T),
-            v0 = randn(T,size(nep,1)),
+            λ0::Number = zero(T),
+            v0::Vector = randn(T,size(nep,1)),
```

Comment:
Please make the input consistent as we agreed in https://github.com/nep-pack/NonlinearEigenproblems.jl/wiki/Style-guidelines-and-notes i.e. λ not λ0 etc.